linq4j/src/main/java/org/apache/calcite/linq4j/tree/MethodDeclaration.java

```diff
@@ -57,20 +60,39 @@ public MethodDeclaration(int modifier, String name, Type resultType,
   }
 
   @Override public void accept(ExpressionWriter writer) {
-    String modifiers = Modifier.toString(modifier);
-    writer.append(modifiers);
+    final ExpressionWriter writerForUnsplitMethod = writer.usesMethodSplitting()
```

Comment:
I have the impression the intent of this line is not very obvious, perhaps a comment would help to clarify it.

Reply:
Done in [dc1c66d](https://github.com/apache/calcite/pull/3901/commits/dc1c66d14b38ad122312b99861fca20794701504)

Reply:
Thanks @jduo , just a small comment, because I find the comment a bit confusing (maybe I'm getting it wrong), where it says:
```
... serialize the method declaration directly to the supplied writer if method splitting is enabled or ...
```
shouldn't it say "disabled" ?

Reply:
Yes that's correct. I've updated this comment, but note that it is now in ClassDeclaration instead of MethodDeclaration. The method splitter requires an enclosing class to write some of its output too (it can extract variable names to instance variables and move blocks of code to generated private methods to reduce the size of a method).